internal/dag/dag.go

```diff
@@ -142,23 +149,35 @@ func (d *MapDag) traceNode(identifier string, tree map[string]Node) error {
 	for _, n := range d.nodes[identifier].Neighbors() {
 		// if we have already visited this neighbor, then we have already
 		// visited its neighbors, so we can skip.
-		if _, ok := tree[n.Identifier()]; ok {
+		identifier := n.Identifier(d.defaultRegistry)
+		if _, ok := tree[identifier]; ok {
 			continue
 		}
-		tree[n.Identifier()] = n
-		if err := d.traceNode(n.Identifier(), tree); err != nil {
+		tree[identifier] = n
+		if err := d.traceNode(identifier, tree); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
 // GetNode returns a node in the dag.
-func (d *MapDag) GetNode(identifier string) (Node, error) {
-	if _, ok := d.nodes[identifier]; !ok {
-		return nil, errors.Errorf("node %s does not exist", identifier)
+func (d *MapDag) GetNode(identifier, reg string) (Node, error) {
+	id := AddDefaultRegistry(identifier, reg)
+	if _, ok := d.nodes[id]; !ok {
+		return nil, errors.Errorf("node %s does not exist", id)
+	}
+	return d.nodes[id], nil
+}
+
+// AddDefaultRegistry adds the default registry to an image if there is no registry.
+func AddDefaultRegistry(image, reg string) string {
```

Comment:
i would imagine there's some logic in [`github.com/google/go-containerregistry/pkg/name`](https://pkg.go.dev/github.com/google/go-containerregistry/pkg/name) that does something similar and is more robust, e.g. handles input/cases we haven't even thought of yet.

Have you already looked into using that package instead of writing our own logic?

An example of where we're already using some of that package can be found in: https://github.com/crossplane/crossplane/blob/master/cmd/crank/beta/trace/internal/printer/default.go#L321-L333